src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -48,6 +50,15 @@ internal static BoundBlock InjectInstrumentation(MethodSymbol method, BoundBlock
                     BoundExpression payloadNullTest = factory.Binary(BinaryOperatorKind.ObjectEqual, boolType, factory.Field(null, payloadField), factory.Null(payloadType));
                     BoundStatement payloadIf = factory.If(payloadNullTest, createPayloadCall);
 
+                    // Methods defined with => syntax don't have a block syntax, and so won't naturally get a sequence point for the
+                    // entry of the method. It is a requirement that there be a sequence point before any executable code. A sequence point
+                    // won't be generated automatically for the synthesized if statement because the if statement is compiler generated, so
+                    // force the generation of a sequence point for the if statement.
+                    if (methodUsesArrowSyntax)
+                    {
+                        payloadIf = factory.SequencePoint(payloadIf.Syntax, payloadIf);
```

Comment:
Doing this unconditionally is wrong. I am sure there are scenarios when we don't create sequence points at all.
Second, it is wrong to create sequence points before LocalRewriter. Third, we probably would have to deal with this specially if instrumentation was a part of the lowering pass. @JohnHamby, if we haven't created an issue to track the move to the infrastructure we discussed last Friday, could you create one?